src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                results.put(indexName, DetectorResult.failedValidation(indexName, e.getMessage()).toMap());
+                processNextIndex(indices, currentIndex + 1, tenantId, maxRetries, results, listener);
+            }
+        });
+    }
+
+    // Flow: get mappings -> LLM generates config -> validate config -> optimize params -> validate model -> create detector
+    private void processSingleIndex(String indexName, String tenantId, int maxRetries, ActionListener<String> listener) {
+        getMappingsAndFilterFields(
+            indexName,
+            ActionListener
+                .wrap(
+                    mappingContext -> generateAndParseConfig(
+                        mappingContext,
+                        tenantId,
+                        maxRetries,
+                        listener,
+                        (suggestions, listenerCallback) -> validateDetectorPhase(
+                            suggestions,
+                            mappingContext,
+                            tenantId,
+                            maxRetries,
+                            0,
+                            listenerCallback
+                        )
+                    ),
+                    listener::onFailure
+                )
+        );
+    }
+
+    private String extractResponseFromDataAsMap(Map<String, Object> dataAsMap) {
+        if (dataAsMap == null) {
+            return null;
+        }
+
+        if (dataAsMap.containsKey("response")) {
+            return (String) dataAsMap.get("response");
+        } else if (dataAsMap.containsKey("output")) {
+            // Parse Bedrock format: output.message.content[0].text
+            try {
+                Map<String, Object> output = (Map<String, Object>) dataAsMap.get("output");
+                Map<String, Object> message = (Map<String, Object>) output.get("message");
+                List<Map<String, Object>> content = (List<Map<String, Object>>) message.get("content");
+                return (String) content.getFirst().get("text");
+            } catch (Exception e) {
+                log.error("Failed to parse Bedrock response format", e);
+                return null;
+            }
+        } else if (dataAsMap.containsKey("content")) {
+            // Parse Claude format: content field as array
+            try {
+                List<Map<String, Object>> content = (List<Map<String, Object>>) dataAsMap.get("content");
+                return (String) content.getFirst().get("text");
+            } catch (Exception e) {
+                log.error("Failed to parse Claude content format", e);
+                return null;
+            }
+        } else {
+            log.error("Unknown response format. Available keys: {}", dataAsMap.keySet());
+            return null;
+        }
+    }
```

Comment:
This hard-coded parsing and relying on exception handling seems like it could be improved.  At least add a comment showing the expected format, so if this breaks in the future, the next maintainer (or future-you) can quickly identify the format change without walking through the code.